source/draft_users.rst

```diff
@@ -0,0 +1,177 @@
+the other endpoints, with certain key differences.
+
+GET /users
+~~~~~~~~~~
+
+Returns a list of all users in the database:
+
+.. code-block:: javascript
+
+    [
+      {
+        "username": "example",
+        "displayname": "X. Ample User",
+        "email": "example@example.com",
+        "active": true,
+        "admin": false,
+        "created_at": "2015-02-29",
+        "deleted_at": null
+      },
+      {
+        ...
+      },
+      ...
+    ]
+
+This endpoint can be accessed by any authenticated user.
+
+GET /users/:username
+~~~~~~~~~~~~~~~~~~~~
+
+Returns a specific user from the database.
+
+.. code-block:: javascript
+
+    {
+      "username": "example",
+      "displayname": "X. Ample User",
+      "email": "example@example.com",
+      "active": true,
+      "admin": false,
+      "created_at": "2015-02-29",
+      "deleted_at": null
+    }
+
+This endpoint can be accessed by any authenticated user, and may be used to look up any
+user.
+
+POST /users
+~~~~~~~~~~~
+
+Create a new user.
+
+Request:
+
+.. code-block:: javascript
+
+    {
+      "username": "example",
+      "password": "password",
+      "displayname": "X. Ample User",
+      "email": "example@example.com"
+    }
+
+Response:
+
+.. code-block:: javascript
+
+    {
+      "username": "example",
+      "displayname": "X. Ample User",
+      "email": "example@example.com",
+      "active": true,
+      "admin": false,
+      "created_at": "2015-02-29",
+      "deleted_at": null
+    }
+
+This endpoint may only be accessed by admins. It is therefore recommended that admins
+provide the user with a temporary password and have the user change the password when
+they log in.
+
+POST /users/:username
+~~~~~~~~~~~~~~~~~~~~~
+
+Update a user's information.
+
+Request:
+
+.. code-block:: javascript
+
+    {
+      "username": "example-user",
+      "password": "new-password",
+      "displayname": "Mr. Example",
+      "email": "examplej@example.com"
+    }
+
+Response:
+
+.. code-block:: javascript
+
+    {
+      "username": "example-user",
+      "displayname": "Mr. Example",
+      "email": "examplej@example.com",
+      "active": true,
+      "admin": false,
+      "created_at": "2015-02-29",
+      "deleted_at": null
+    }
+
+This endpoint may be accessed by admins or the user who is being updated. However, the
+``admin`` field may only be set by an admin.
+
+DELETE /users/:username
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Delete a user. Returns a 200 OK with empty response body on success, or an
+:ref:`error<draft_errors>` on failure. Only accessible to admins.
+
+---------------
+
+Role Management
+---------------
+
+Role management is handled through the ``projects`` endpoints. The projects model contains
+a ``users`` object, which contains three lists: ``members``, ``spectators``, and
+``managers``, each lists of usernames. An admin or project manager may set these at any
+time, adding to or removing from any of the lists.
```

Comment:
A manager may add and remove members, spectators, and managers to/from the project, including themself.